File: external/cec2006_src/cec2006-benchmarks-master/code/matlab/g02.m
function f = g02_objective(x)
% CEC 2006 g02 constrained optimization problem objective function.
%
% Parameters:
% x: Decision variables [x1, x2, ..., x20]
%
% Returns:
% f: Objective function value

term1 = sum(cos(x).^4);
term2 = 2 * prod(cos(x).^2);
term3 = sqrt(sum((1:20)' .* x.^2));
f = -abs((term1 - term2) / term3);

function g = g02_constraints(x)
% CEC 2006 g02 constraints.
%
% Parameters:
% x: Decision variables [x1, x2, ..., x20]
%
% Returns:
% g: Constraint values (should be <= 0 for feasible)

g = zeros(2, 1);
g(1) = 0.75 - prod(x);
g(2) = sum(x) - 7.5 * 20;

function bounds = g02_bounds()
% Returns the bounds for g02 problem.
%
% Returns:
% bounds: [lower, upper] for each variable

bounds = repmat([0, 10], 20, 1);

% Example usage
if ~exist('x', 'var')
    % Example solution (best known)
    x_opt = zeros(20, 1);
    f_opt = g02_objective(x_opt);
    g_opt = g02_constraints(x_opt);
    
    fprintf('Objective value: %f\n', f_opt);
    fprintf('Constraints: ');
    disp(g_opt');
    fprintf('All constraints satisfied: %d\n', all(g_opt <= 0));
end

File: external/cec2006_src/cec2006-benchmarks-master/code/python/g02.py
import numpy as np

def g02_objective(x):
    """
    CEC 2006 g02 constrained optimization problem objective function.
    
    Parameters:
    x (array-like): Decision variables [x1, x2, ..., x20]
    
    Returns:
    float: Objective function value
    """
    term1 = sum(np.cos(xi)**4 for xi in x)
    term2 = 2 * np.prod(np.cos(xi)**2 for xi in x)
    term3 = np.sqrt(sum(i * xi**2 for i, xi in enumerate(x, 1)))
    return -abs((term1 - term2) / term3)

def g02_constraints(x):
    """
    CEC 2006 g02 constraints.
    
    Parameters:
    x (array-like): Decision variables [x1, x2, ..., x20]
    
    Returns:
    list: Constraint values (should be <= 0 for feasible)
    """
    g = []
    g.append(0.75 - np.prod(x))
    g.append(sum(x) - 7.5 * 20)
    return g

def g02_bounds():
    """
    Returns the bounds for g02 problem.
    
    Returns:
    list: [(lower, upper), ...] for each variable
    """
    return [(0, 10)] * 20

# Example usage
if __name__ == "__main__":
    # Example solution (best known)
    x_opt = [0] * 20  # Approximate
    f_opt = g02_objective(x_opt)
    g_opt = g02_constraints(x_opt)
    
    print(f"Objective value: {f_opt}")
    print(f"Constraints: {g_opt}")
    print(f"All constraints satisfied: {all(g <= 0 for g in g_opt)}")